controllers/ratesController.js

```diff
@@ -0,0 +1,107 @@
+const config = require('../config')
+const helpers = require('../helpers')
+
+const logger = helpers.logger
+
+/**
+* Parse 'from' and 'to' from the req
+*/
+const parseFromTo = (req, res) => {
+    if(typeof req.params.from !== 'string' || typeof req.params.to !== 'string') {
+        logger.error('invalid input for action')
+        res.status(400).send('invalid input for action')
+        return {valid: false}
+    }
+    return {valid: true, from: req.params.from.toUpperCase(), to: req.params.to.toUpperCase()}
+}
+
+/**
+* GET action for get specific rate info
+*/
+const getRate = async (req, res) => {
+    
+    //validate and parse input
+    const parsedFromTo = parseFromTo(req, res)
+    if(!parsedFromTo.valid) return
+
+    let parsedTimeframe = parseInt(req.query.timeframe)
```

Comment:
Instead of a method called `parseFromTo` you can have a `parseParams` and include this one too

Reply:
done